### PR TITLE
Performance optimizations cleanup

### PR DESCRIPTION
**Pull request by luontola**
Follow up from #1114 code review.

### CODE REVIEW

File: src/clj/rems/api/applications_v2.clj [hunk 1]
```diff
@@ -129,13 +143,16 @@
        (filter own-application?)))
 
 (defn reload-cache! []
+  ;; TODO: Here is a small chance that a user will experience a cache miss. Consider rebuilding the cache asynchronously and then `reset!` the cache.
   (events-cache/empty! all-applications-cache)
   (refresh-all-applications-cache!))
 
 ;; empty the cache occasionally in case some of the injected entities are changed
 (mount/defstate all-applications-cache-reloader
+  ;; TODO: consider refactoring this mount-cron-thing which is duplicated in a few places
```
Comment (opqdonut) [inline src/clj/rems/api/applications_v2.clj]: yeah please do that

File: src/clj/rems/api/applications_v2.clj [hunk 2]
```diff
@@ -98,6 +111,7 @@
    (fn [state events]
      ;; Because enrich-with-injections is not idempotent,
      ;; it's necessary to hold on to the "raw" applications.
+     ;; TODO: consider making enrich-with-injections idempotent (move dissocs to hide-non-public-information and other small refactorings)
```
Comment (opqdonut) [inline src/clj/rems/api/applications_v2.clj]: please try this out, perhaps in a follow-up PR

File: src/clj/rems/db/dynamic_roles.clj [hunk 3]
```diff
@@ -42,15 +42,22 @@
...
 (defn get-roles [user]
   (->> (events-cache/refresh!
         dynamic-roles-cache
         (fn [state events]
-          (let [apps (reduce permissions-of-all-applications (::apps state) events)]
-            {::apps apps
-             ::roles-by-user (group-roles-by-user apps)})))
+          (let [permissions (reduce permissions-of-all-applications (::permissions state) events)]
+            {::permissions permissions
+             ::roles-by-user (group-roles-by-user permissions)})))
```
Comment (opqdonut) [inline src/clj/rems/db/dynamic_roles.clj]: is `group-roles-by-user` really so slow that we need to memoize it?

Comment (luontola) [reply src/clj/rems/db/dynamic_roles.clj]: There are benchmark measurements in the commit message 87a598bafa9d210b31d690e70d85f8932a370615. It used to be a linear search which calls `apply-user-permissions` for every application, but now it's just a hashmap lookup. Filtering a person's own applications and open/handled reviews is still a linear search, but most users can see only a few applications so it's less of an issue.

File: src/clj/rems/api/applications_v2.clj (hunk 1 above)
Comment (luontola) [reply src/clj/rems/api/applications_v2.clj]: Done in #1138

**Pull request merged**